addons/survey/models/survey_user_input.py

```diff
@@ -438,7 +438,8 @@ def _prepare_statistics(self):
 
             question_section = question.page_id.title or _('Uncategorized')
             for user_input in self:
-                user_input_lines = user_input.user_input_line_ids.filtered(lambda line: line.question_id == question)
+                user_input_lines = user_input.user_input_line_ids.filtered(lambda line:
+                    line.question_id == question and not (line.answer_type == 'char_box' and not question.comment_count_as_answer))
```

Comment:
Hello @julien-banken :smile: 

Actually here we loop on the `user_input` so on one user survey entry at a time. :slightly_smiling_face: 
We also filter the `user_input_lines` here by question (` line.question_id == question`).

This means that, if the question is a multiple choice one, we could get multiple user input lines, but if the question is a simple choice or another type we'll always get only one user input line because the user can only return one answer to the question :slightly_smiling_face: 
You can see that with the implementation of the ` _simple_question_answer_result` method called for the date / datetime / numerical box types underneath (the scored question types).
```python
answer_result_key = self._simple_question_answer_result(user_input_lines)
...
def _simple_question_answer_result(self, user_input_line):
    if user_input_line.skipped:
        return 'skipped'
    elif user_input_line.answer_is_correct:
        return 'correct'
    else:
        return 'incorrect'
```
We pass the `user_input_lines` but concretely we'll always get one user input line :slightly_smiling_face: 

The issue we had here was that we were waiting for one user input line for simple choice questions but we got multiple because we forgot to exclude the comments not counted as answers.

If I missed something don't hesitate :slightly_smiling_face: 
Cheers!

Reply:
Ok, I got it: In the loop, `user_input_lines` is the user answer for the considered question. For the simple choice question, we should only get one answer. It would not make sens to adapt the `_simple_choice_question_answer_result` method to support multiple answers.

Thank you for explanation :+1:

Reply:
Exactly! Your explanations are way better than mine! :joy:

Reply:
I have an old :brain: so can't we rewrite a bit the condition with less not + parenthesis ? As not (A and B) is not(A) or not(B) (if I remember well):
```suggestion
                    line.question_id == question and (line.answer_type != 'char_box' or question.comment_count_as_answer)))
```
Looking at it, wondering if it is really what we want :coffee: don't we want to exclude char_box that are comments ?

Reply:
Hello @tde-banana-odoo :slightly_smiling_face: 
cc @julien-banken 
Updated :slightly_smiling_face: It should now be the right condition if I get it right :smiling_face_with_tear: 
Thanks!